core/src/fnc/script/from.rs

```diff
@@ -22,6 +25,60 @@ fn check_nul(s: &str) -> Result<(), Error> {
 	}
 }
 
+fn try_object_to_geom(object: &Object) -> Option<Geometry> {
+	if object.len() != 2 {
+		return None;
+	}
+
+	let Some(Value::Strand(Strand(key))) = object.get("type") else {
+		return None;
```

Comment:
I meant doing

```rust
let Value::Strand(Strand(key)) = object.get("type")?;
```

instead of

```rust
let Some(Value::Strand(Strand(key))) = object.get("type") else {
		return None;
}
```

The `?` will propagate the `None` for you so you don't need `return None;` here.

Reply:
> let Value::Strand(Strand(key)) = object.get("type")?;

This doesn't work. The pattern is not exhaustively matching all variants of Value. So you need an else clause.